ShopAppTests/App/Account/ViewModels/BaseAddressListViewModelSpec.swift

```diff
@@ -0,0 +1,201 @@
+//
+//  BaseAddressListViewModelSpec.swift
+//  ShopAppTests
+//
+//  Created by Evgeniy Antonov on 3/1/18.
+//  Copyright © 2018 RubyGarage. All rights reserved.
+//
+
+import Nimble
+import Quick
+import RxSwift
+import ShopApp_Gateway
+
+@testable import ShopApp
+
+class BaseAddressListViewModelSpec: QuickSpec {
+    override func spec() {
+        var viewModel: BaseAddressListViewModel!
+        var customerUseCaseMock: CustomerUseCaseMock!
+        var deleteAddressUseCaseMock: DeleteAddressUseCaseMock!
+        var updateDefaultAddressUseCaseMock: UpdateDefaultAddressUseCaseMock!
+        
+        beforeEach {
+            let authenticationRepositoryMock = AuthentificationRepositoryMock()
+            customerUseCaseMock = CustomerUseCaseMock(repository: authenticationRepositoryMock)
+            let paymentsRepositoryMock = PaymentsRepositoryMock()
+            updateDefaultAddressUseCaseMock = UpdateDefaultAddressUseCaseMock(repository: paymentsRepositoryMock)
+            deleteAddressUseCaseMock = DeleteAddressUseCaseMock(repository: paymentsRepositoryMock)
+            viewModel = BaseAddressListViewModel(customerUseCase: customerUseCaseMock, updateDefaultAddressUseCase: updateDefaultAddressUseCaseMock, deleteAddressUseCase: deleteAddressUseCaseMock)
+        }
+        
+        describe("when view model initialized") {
+            it("should have variables with correct initial values") {
+                expect(viewModel.selectedAddress).to(beNil())
+            }
+        }
+        
+        describe("when data loaded") {
```

Comment:
There are few `it` closures but we need few `context` closures (with one `it` closure in each) because there are two different api response states (success and error). You need to change it in all test cases in this class.